monai/transforms/intensity/array.py

```diff
@@ -1813,9 +1813,10 @@ def _set_default_range(self, img: NdarrayOrTensor) -> Sequence[Sequence[float]]:
         n_dims = len(img.shape[1:])
 
         k = self.shift_fourier(img, n_dims)
+        k, *_ = convert_data_type(k, output_type=np.ndarray)
```

Comment:
May I know why convert `k` to numpy array here?

Thanks.

Reply:
If k is torch.tensor in gpu device, this line: https://github.com/Project-MONAI/MONAI/pull/3871/files#diff-83ae933a39467d71ab978c0bba9debaa2db45f24a74c127a664f1044fc0017dfR1784 will raise an error.

In addition, I think the generated intensity range should has the same type as the argument `intensity_range`:
Sequence[Union[Sequence[float], float]]

Reply:
Thanks @Nic-Ma , I submitted a new commit, and put the convert step after calculation.